homeassistant/components/met/weather.py

```diff
@@ -142,23 +187,31 @@ def pressure(self):
     @property
     def humidity(self):
         """Return the humidity."""
-        return self.coordinator.data.current_weather_data.get("humidity")
+        return self.coordinator.data.current_weather_data.get(
+            ATTR_MAP[ATTR_WEATHER_HUMIDITY]
+        )
 
     @property
     def wind_speed(self):
         """Return the wind speed."""
-        speed_m_s = self.coordinator.data.current_weather_data.get("wind_speed")
-        if self._is_metric or speed_m_s is None:
-            return speed_m_s
+        speed_km_h = self.coordinator.data.current_weather_data.get(
+            ATTR_MAP[ATTR_WEATHER_WIND_SPEED]
+        )
+        if self._is_metric or speed_km_h is None:
+            return speed_km_h
 
-        speed_mi_s = convert_distance(speed_m_s, LENGTH_METERS, LENGTH_MILES)
-        speed_mi_h = speed_mi_s / 3600.0
+        speed_mi_h = convert_distance(speed_km_h, LENGTH_KILOMETERS, LENGTH_MILES)
         return int(round(speed_mi_h))
 
     @property
     def wind_bearing(self):
         """Return the wind direction."""
-        return self.coordinator.data.current_weather_data.get("wind_bearing")
+        degrees = self.coordinator.data.current_weather_data.get(
+            ATTR_MAP[ATTR_WEATHER_WIND_BEARING]
+        )
+        if degrees is None:
+            return
+        return compass_degrees_to_code(degrees)
```

Comment:
I think we can keep returning degrees here to avoid another breaking change. The frontend should cope and convert it if I'm reading the frontend correctly. Please check this.

Reply:
Thanks for the heads up, hadn't looked at the front end code. I'll Check tonight and update accordingly.

Reply:
The frontend does indeed perform this conversion. I've rolled this back.